src/Symfony/Component/Security/Http/Firewall/LogoutListener.php

```diff
@@ -61,6 +63,16 @@ public function __construct(TokenStorageInterface $tokenStorage, HttpUtils $http
     }
 
     /**
+     * Sets a request matcher.
+     *
+     * @param RequestMatcherInterface $requestMatcher
+     */
+    public function setRequestMatcher(RequestMatcherInterface $requestMatcher)
```

Comment:
Cannot it be passed as a new optional constructor argument rather than by using a method call?

I'm not sure someone is supposed to replace the abstract service definition, so it shouldn't be an issue.